$NetBSD$

Reply-To: namazu-devel-ja@namazu.org
Subject: Re: $conf::EXCLUDE_PATH, ($File::Find::prune = 1 ;)
To: namazu-devel-ja@namazu.org
Message-Id: <20000327154303L.makoto@ki.nu>
In-Reply-To: <20000325174240Q.makoto@ki.nu>
References: <20000325174240Q.makoto@ki.nu>
X-ML-Name: namazu-devel-ja
X-Mail-Count: 00274
X-MLServer: fml [fml 3.0pl#17]; post only (anyone can post)
X-ML-Info: If you have a question, send e-mail with the body
	"help" (without quotes) to the address namazu-devel-ja-ctl@namazu.org;
	help=<mailto:namazu-devel-ja-ctl@namazu.org?body=help>
X-Mailer: Mew version 1.94 on Emacs 20.5 / Mule 4.0
 =?iso-2022-jp?B?KBskQjJWMWMbKEIp?=
X-Face: %{9B8A;;46J-!*nx-Cd8z*g}rH'x8liNOwWoSCEFDFe.+"~zj32d0g0,$y[Z:aX7Y^U&t5*
 lGzCI]kY<E*I"_ANjH9I1X|x@QD}QFEgU&\*]QKBlRo#Vu/7_~?lMzz\LbYgVO[?0|O{wn/I2"o4,=
 (jU;9M^2#Fv=yd#U@U=B\//jfR5#,Q"MhvSsx?b/Idp.[2La+1VH#Y|?gdw&O/|7GM.tP#_#1,0/90
 3!?,BAMY\-
X-Prom-Mew: Prom-Mew 1.94 (procmail reader for Mew)
X-PGP-Fingerprint: 82 AD 5D F4 1E DC 00 BF  11 90 35 83 ED 85 6F D6
X-URL: http://www.ki.nu/software/namazu/tutorial.shtml
X-Dispatcher: imput version 990905(IM130)
Mime-Version: 1.0
Content-Type: Text/Plain; charset=iso-2022-jp
Content-Transfer-Encoding: 7bit
Precedence: bulk
Lines: 68
Sender: namazu-devel-ja-admin@namazu.org

>                                            千葉市中央区長洲
>                                                    藤原  誠
namazu-devel-ja: 00271 にて
makoto> で、同じことを mknmz.in に施して見たのですが、closure の中では効か
makoto> ないのでしょうか。(これではだめでした)

$File::Find::prune = 1 とするのは directory を見つけた時である必要がある
のですね。

○ 使用前
@@ find_target finished: Mon Mar 27 15:31:34 2000
@@ Target Files: 1249 (Scan Performance: Elapsed Sec.: 38, Files/sec: 32.9)
@@   Possible: 2688, Not allowed: 0, Denied: 242, Excluded: 1197
@@   MTIME too old: 0, MTIME too new: 0
1249 files are found to be indexed.
1/1249 - /amd/milano/export/home/htdocs/software/DeleGate.shml [text/plain]

○ 使用後
@@ find_target finished: Mon Mar 27 15:29:32 2000
@@ Target Files: 1249 (Scan Performance: Elapsed Sec.: 21, Files/sec: 59.5)
@@   Possible: 1499, Not allowed: 0, Denied: 242, Excluded: 8
@@   MTIME too old: 0, MTIME too new: 0
1249 files are found to be indexed.
1/1249 - /amd/milano/export/home/htdocs/software/DeleGate.shml [text/plain]

変更内容

--- mknmz.in-o	Wed Mar 22 23:30:02 2000
+++ ./scripts/mknmz.in	Mon Mar 27 15:36:28 2000
@@ -1290,6 +1290,15 @@
 	return;   # skip a file name containing LF/CR/TAB chars.
     }
 
+    if (defined $conf::EXCLUDE_PATH && 
+	$target =~ /$conf::EXCLUDE_PATH/ ) 
+    {
+        util::vprint(sprintf(_("Excluded:	%s"), $target));
+	$counts_ref->{'possible'}++;
+	$counts_ref->{'excluded'}++;
+	  $File::Find::prune = 1;
+	  return; 
+    }
     return unless -f $target;  # Only file is targeted.
 
     $counts_ref->{'possible'}++;
@@ -1300,14 +1309,6 @@
 	return;
     }
 
-
-    if (defined $conf::EXCLUDE_PATH && 
-	$target =~ /$conf::EXCLUDE_PATH/ ) 
-    {
-        util::vprint(sprintf(_("Excluded:	%s"), $target));
-	$counts_ref->{'excluded'}++;
-	return; 
-    }
 
     #
     # Do processing just like find's  --mtime option.

Excluded: を file と directory に分けた方が親切かも知れない
と思っています。

# きょう 3/27 - 3/31 まで留守にします。
---
(藤原)
